eodatasets/prepare/ncep_reanalysis_surface_pr_wtr.py

```diff
@@ -0,0 +1,166 @@
+"""
+Create index files from the command-line
+"""
+
+import datetime
+import uuid
+import urllib.parse
+from pathlib import Path
+
+import click
+import rasterio
+import rasterio.crs
+
+from eodatasets.serialise import write_yaml_from_dict
+from .utils import read_paths_from_file
+
+
+NOAA_WATER_VAPOUR_NS = uuid.UUID(hex='857bd048-8c86-4670-a2b4-5dbea26d7692')
+NOAA_DESCRIPTION = ''
+
+
+def get_coords(collection, xkey='lon', ykey='lat'):
+    """
+    transforms the bounds of a collection into an
+    encompassing bounding box. Keys for x and y are
+    customisable
+    """
+    return {
+        'll': {
+            ykey: collection.bounds.bottom,
+            xkey: collection.bounds.left
+        },
+        'lr': {
+            ykey: collection.bounds.bottom,
+            xkey: collection.bounds.right
+        },
+        'ul': {
+            ykey: collection.bounds.top,
+            xkey: collection.bounds.left
+        },
+        'ur': {
+            ykey: collection.bounds.top,
+            xkey: collection.bounds.right
+        }
+    }
+
+
+def get_uuid(collection, idx):
```

Comment:
These parameters were unclear to me without finding usages of the method. Maybe rename or add type decorators, or both?
```suggestion
def get_uuid(dataset: DatasetReader, band_index: int):
```

(I'm biased towards types because they make many tools more useful...)